### ISSUE TITLE
Poorly balanced water from cactus recipe.

### ISSUE THREAD

**Issue by Caiggas** [GTNewHorizons/GT-New-Horizons-Modpack#5996]
#### Which modpack version are you using?
2.0.8.8
#
#
#### What did you try to do, and what did you expect to happen?
N/A
#
#### What happened instead? (Attach screenshots if needed)
N/A
#
The above two questions don't really work in this context, sorry. This is my first issue, so I'm sorry if I'm doing this wrong. Anyway:

I am currently playing through the stone age (third run), and I was reminded how utterly useless the cactus to water recipe is. It's so much easier to just pick up a water block that no one will bother to setup a cactus farm for such high effort and low throughout water generation. Worst of all, it's a noob trap. Someone new to the pack and not thoroughly thinking it through may naively assume that it's worth using this recipe simply because it is available and mentioned in the quest book.

#### What do you suggest instead/what changes do you propose?
There are a few solutions I thought of:

1: Disable recipe and remove relevent quest entry. This would prune one unused recipe and get rid of one noob trap.

2: Oredict the cactus water object to water buckets/cells. This does have the added bonus (or overbuff, depending on your opinion I guess) of making early game water recipe crafting way less physically painful (carpal tunnel) and tedious.

 This may have consequences that I haven't heard of, so solution may have to be modified.

If it is too OP, there are a few solutions.

2A: Make the cactus require some basic stone age processing to get the cactus water object. Mallet, knife, furnace smelting, ect.

2B:  --Make a new item which requires 2 or 4 of the cactus water items to craft. Make this item the one oredicted to water buckets/cells. I don't particularly like the solution, but it's better than having to physically place a bucket for each craft operation.--
EDIT: bad idea. Rather, just to make the water objects require several cactuses in a shapeless recipe. There's no point in adding a whole new item.

2C: If making water from cactuses without using a bucket is just not an option, then at least let them be one-to-one crafted with Pam's water barrels. Then at least they can be useful for something.
#

Thank you all for your time. as I said before, this is my first issue. Please let me know if I'm doing anything wrong.

**Comment by richardhendricks** [comment]
The problem is that while hard for early stages, it can become trivial to setup a cactus farm to automate getting fairly large amounts of cactus. The hand recipe is more for emergency amounts of water or for meme-running in a desert. Once a player actually gets to steam and has machines, a lot more water can be extracted per cactus block.

1. No, it is a useful recipe for early game
2. This is way to OP
2A. No
2B No
2C This would be acceptable. Perhaps with a mallet, let it turn into Pam's buckets. This would make crafting foods easier early game, which I am OK with.

**Comment by Caiggas** [comment]
Thank you for your feedback. I have a few questions to ask in reply.

1: I guess I can see how the recipe benefits desert runs, so I agree outright removing it is a bad idea. I'm a bit skeptical of an emergency situation where you have a bucket and some cactus at the same time and can't run ~100 blocks to the nearest river instead. However, the options in 2 wouldn't negate either of these uses.

2: I will defer to your judgement here. I was afraid of it being too OP, at least in it's simplest implimentation.

2a: see below.

2b: Is this unacceptable because the ratio of cactus to water is bad, or that it gets rid of the bucket?
For the first, ratio could be set to anything, even 3x3 instead of 8 around a bucket. 
For the second:
If getting portable water pre-bucket is an issue, the quests give you a bucket before you ever need water, so it's not breaking any gateing. 
If making water without having to place the bucket in the grid is the issue, why?
All it does is require more repetitive physical hand movement to craft. You always do the crafting next to a river, so it's not slowing down your water gather speed. 
In fact, cactus farming slows you down since you have to wait for them to grow, at any cactus/water ratio. 
The cactus advantage is trading total water throughput speed for less repetitive physical hand movement.
In other words, actual balance isn't changing. In fact, you can negatively skew the balance even further.
The advantage you get is strictly IRL. For those of us with carpal tunnel or other nerve issues, it would be a godsend worth sacrificing total crafting throughput.

Having said all this, I just realized that my title and original argument are wrong. The real issue isn't balance. It's physically painful crafting. Huh.

2c: Yeah, looks like we agree. Nothing to add here.

Note: I want to clarify that I mean no offense arguing with your points. I just wanted to further discuss them. Thank you for your time.

**Comment by richardhendricks** [comment]
It isn't too hard to run through a desert gathering sand and cactus and setting up an automated cactus farm is pretty trivial with string and water. Some players may be on a mountain top or other desirable place and need a way to make water. Also consider that cactus stacks in inventory, so 64 cactus = 8 buckets of water, which isn't bad. You could run and collect sand for grout/glass/etc and scoop up a bunch of water too.

2B I would rather change the ratio to 4:1 instead of creating a limited single use item cluttering up the recipe map. 

The real issue is moving the bucket, I guess? Later you get a Forestry Worktable which helps greatly. But before that point, yes it would be annoying. GTNH is suffering.

I guess as a "noob trap" it is a valid complaint.  Wording could be added to the quest that it is a short time/emergency water source?

Changing it to 4:1 is acceptable, and making a 2:1 recipe for pam's water bucket could also be done. It would make cactus farming a viable less annoying source for pam's water than using buckets, but I guess that's not necessarily a bad thing.

Maintaining a meta balance is really hard, it's not surprising that it changes over time as people start using alternative methods.  I guess with the stirling water generator it is pretty easy now to get early water in quantity. That could use a rebalancing too IMO.

**Comment by Prunebane** [comment]
If it's really about the bucket discomfort being the issue, is there a possibility of making a sort of "one-time use", say, _spongy wet glob_ that acts as a (stackable? to-4-stackable?) water bucket, except it's consumed on use? (you'd still need a bucket to pick it up afterwards)
You could also "oredict" it instead of a water bucket for crafting purposes, perhaps. (Could introduce an issue with how buckets are normally handled; I'd tread with care)
This wouldn't mess with the 8:1 balance at all, simply providing a more comfortable way of working with what would be an emergency source of water, and for the sake of not breaking OG recipes, could go with the current recipe sans bucket -> _spongy wet glob_
It would, however, slightly mess with how good the forestry worktable is for handling recipes with water early on. Having a (somewhat) stackable water bucket would mean less need for an actual worktable

**Comment by richardhendricks** [comment]
I like the idea of a 4:1 cactus-to-water change, and a 2:1 for pam's water bucket for early food crafting.

**Comment by Caiggas** [comment]
So, how does this sound: 

New item: _relavent name_
Crafted with 4 cactus (or 4 of the current cactus water item?)
Effectively a water cell.
Cannot place a water block in world?

Current water bucket recipe left alone, to provide an option to get water in a bucket so that it can be placed.

**Comment by Caiggas** [comment]
Actually, adding a new item is clumsy. Just make the existing water object effectively a water cell (no placing in world). Change it's recipe to shapeless 4 cactus. 

Change the current recipe to shapelessly craft one or 2 of the water objects with a bucket to get a filled bucket.

Lastly, add a recipe to craft one of the water objects to two Pam's water barrels. (Actually, would this automatically work because of Oredict?)

**Comment by richardhendricks** [comment]
Yeah I am generally against adding new items. We already have the cactus water drop, let's use that instead. 
I'm also against oredicting, as we've seen that's buggy as shit. Better to just have a recipe that converts the cactus water drop to a Pam's water bucket.

**Comment by Caiggas** [comment]
Wouldn't making the cactus water object work like a water cell require oredicting it to water cells anyway? I apologize if I'm just ignorant about how it works. I only have a general outside understanding of how this stuff works.
